framework/src/packets/ethernet.rs

```diff
@@ -123,14 +123,17 @@ impl fmt::Display for EtherType {
             match *self {
                 EtherTypes::Ipv4 => "IPv4".to_string(),
                 EtherTypes::Ipv6 => "IPv6".to_string(),
-                _ => format!("0x{:04x}", self.0),
+                _ => {
+                    let t = self.0;
+                    format!("0x{:04x}", t)
```

Comment:
huh, why does this have to change?

Reply:
Due to safe packed borrows feature as well. This is the common fix

Reply:
is it because that's a `u16`? cuz `ProtocolNumber` which is a u8 is fine. Not sure I really understand what's actually going on. oh well.